libfreeipmi/driver/ipmi-inteldcmi-driver.c

```diff
@@ -444,6 +445,19 @@ ipmi_inteldcmi_ctx_io_init (ipmi_inteldcmi_ctx_t ctx)
       goto cleanup;
     }
 
+  flags = fcntl(ctx->device_fd, F_GETFD);
+  if (flags == -1)
+    {
```

Comment:
POSIX tells that upon successful completion of F_GETFD the return value shall not be negative. And in case of error fcntl returns -1. So, in case of F_GETFD it is absolutely correct to check return value as < 0

But in case of F_SETFD the successful return value is just "Value other than -1." So, although it is pretty unlikely that any system returns negative value upon success, checking < 0 here is not strictly POSIX complaint. 

see http://pubs.opengroup.org/onlinepubs/9699919799/